karapace/typing.py

```diff
@@ -59,23 +59,47 @@ class Mode(StrEnum):
     readwrite = "READWRITE"
 
 
-class Version(int):
+class Version:
     LATEST_VERSION_TAG: ClassVar[str] = "latest"
     MINUS_1_VERSION_TAG: ClassVar[int] = -1
 
-    def __new__(cls, version: int) -> Version:
+    def __init__(self, version: int) -> None:
         if not isinstance(version, int):
             raise InvalidVersion(f"Invalid version {version}")
-        if (version < cls.MINUS_1_VERSION_TAG) or (version == 0):
+        if (version < Version.MINUS_1_VERSION_TAG) or (version == 0):
             raise InvalidVersion(f"Invalid version {version}")
-        return super().__new__(cls, version)
+        self._value = version
 
     def __str__(self) -> str:
-        return f"{int(self)}"
+        return f"{int(self._value)}"
 
     def __repr__(self) -> str:
-        return f"Version={int(self)}"
+        return f"Version({int(self._value)})"
+
+    def __gt__(self, other: Version) -> bool:
+        return self._value > other.value
+
+    def __ge__(self, other: Version) -> bool:
+        return self._value >= other.value
+
+    def __lt__(self, other: Version) -> bool:
+        return self._value < other.value
+
+    def __le__(self, other: Version) -> bool:
+        return self._value <= other.value
+
+    def __eq__(self, other: object) -> bool:
+        if isinstance(other, Version):
```

Comment:
**[non-blocking]**
Would be nice to also have this check on the other comparison methods.

Reply:
Done.

Reply:
@jjaakola-aiven think you have to either raise NotImplementedError or return NotImplemented

Reply:
ah Anton was faster :-)